lib/client.js

```diff
@@ -590,10 +593,25 @@ Request.prototype._getFormData = function(){
  */
 
 Request.prototype.callback = function(err, res){
+  // console.log(this._retries, this._maxRetries)
+  if (this._maxRetries && this._retries++ < this._maxRetries && shouldRetry(err, res)) {
+    var timeout = 0;
+    var responseTimeout = 0;
+    if (err && err.timeout) {
+      if (err.message.indexOf('Response') == 0) {
```

Comment:
Yeah, this is really ugly because `_timeout` and `_responseTimeout` are zeroed out in `clearTimeout`, before `callback` is called. Any reason we can't just delete [L45-L46](https://github.com/visionmedia/superagent/blob/master/lib/request-base.js#L45-L46) in `request-base`?